onebone/signal/hampel_filter.py

```diff
@@ -0,0 +1,106 @@
+    If the MAD > 3 * sigma condition is satisfied,
+    the value is replaced with the median value.
+
+    .. math::
+    Local\,median\,:\,m_i = median(x[window\:\_\: size])\\
+    Median\,absolute\,deviation\,:\,MAD_i = median(\left|x_i-k-m_i\right|,...\:,\left|x_i+k-m_i\right|)).
+    '''
+    MAD scale estimate,
+    defined as : S_k = 1.4826 * median_j∈-K,K]{|x_(k-j) - m_k|}.
+
+    '''
+
+    Parameters
+    ----------
+    x : numpy.ndarray
+        1d-timeseries data
+    window_size : int,
+        Lenght of the sliding window.
+        Only integer types are available,
+        and the window size must be adjusted according to your data.
+    n_sigma : float, defalut=3
+        Coefficient of standard deviation.
+
+    Returns
+    ----------
+    filtered_series : numpy.ndarray
+        A value from which Outlier or NaN has been removed by the filter.
+    index : list
+        Returns the index corresponding to Outlier.
+
+    References
+    ----------
+    [1] Pearson, Ronald K., et al. "Generalized hampel filters."
+    EURASIP Journal on Advances in Signal Processing 2016.1 (2016): 1-18.
+    DOI: 10.1186/s13634-016-0383-6
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> fs = 100.0
+    >>> t = np.linspace(0, 1, int(fs))
+    >>> y = np.sin(2 * np.pi *10.0 * t)
+    >>> np.put(y,[9, 13, 24, 30, 45, 51,78],4)
+    >>> first_hampel_filter_array = hampel_filter.hampel_filter(y,2)[0]
+    >>> second_hampel_filter_array = hampel_filter.hampel_filter(y,3)[0]
+
+    .. image:: https://bit.ly/3t55Dlc #nopa
+        :width: 300
+
+    """
+
+    k = 1.4826  # scale factor for Gaussian distribution
+    # The factor 1.4826 makes the MAD scale estimate an unbiased estimate
+    # of the standard deviation for Gaussian data.
+
+    # Check inputs
+    if not isinstance(x, np.ndarray):
+        raise TypeError("'series' must be np.ndarray")
+    if not isinstance(window_size, int):
+        raise TypeError("'window_size' must be integer")
+    if not isinstance(n_sigmas, (int, float)):
+        raise TypeError("'n_sigmas' must be int or float")
+
+    copy_series = x.copy()
+    oulier_index = []
+
+    ## define sliding window
+    indexer = (
+        np.arange(window_size)[None, :] + np.arange(int(len(copy_series) - window_size))[:, None]
+    )
+
+    ## define window median
+    window_median = np.median(copy_series[indexer], axis=1)
+    window_median_array = np.repeat(window_median, window_size, axis=0).reshape(
+        np.shape(copy_series[indexer])[0], window_size
+    )
+
+    ## get MAD * k
+    MAD_k = k * np.median(np.abs(copy_series[indexer] - window_median_array), axis=1)
+
+    ## get comparative value
+    value = np.abs(copy_series[indexer][:, 0] - window_median)
+
+    filtered_series = np.where(value > n_sigmas * MAD_k, window_median, copy_series[indexer][:, 0])
+
+    ## get index
+    oulier_index = np.where(value < n_sigmas * MAD_k, None, indexer[:, 0])
```

Comment:
[100 line](https://github.com/Onepredict/onebone/pull/85/files#diff-4ce7d90e8f2fcd6a396e0e44cf2a9003cf1aefbc0b6c3f64dff040f68a8208d6R100)에서 `value > n_sigmas * MAD_k` 일때 outlier로 판단하여 필터링하므로, 아래와 같이 변경해야 할 것 같습니다.
```suggestion
    oulier_index = np.where(value <= n_sigmas * MAD_k, None, indexer[:, 0])
```

Reply:
> [100 line](https://github.com/Onepredict/onebone/pull/85/files#diff-4ce7d90e8f2fcd6a396e0e44cf2a9003cf1aefbc0b6c3f64dff040f68a8208d6R100)에서 `value > n_sigmas * MAD_k` 일때 outlier로 판단하여 필터링하므로, 아래와 같이 변경해야 할 것 같습니다.

엇 맞네요 감사합니다!!